runtime.md

```diff
@@ -33,24 +33,79 @@ This is provided so that consumers can find the container's configuration and ro
 }
 ```
 
-## Lifecycle
+## Operations
+
+OCI compliant runtimes MUST support the following operations, unless support for the operation can not be supported by the base operating system (kernel) itself.
+
+OPEN: we should consider specifying the options/args for each operation.
+For example, checkpoint (for runc) seems to have a lot of params.
 
 ### Create
 
-Creates the container: file system, namespaces, cgroups, capabilities.
+Using the `config.json` and `runtime.json` files, along with the file system path referenced in the `config.json`, this operation MUST create a new container.
+This includes creating the relevant namespaces, cgroups and configuring the appropriate capabilities for the container.
+
+### Start
+
+This operation MUST start a previously created container.
+Starting a container MUST create a new process, as specified by the `config.json` file within the scope of the container.
+Attempting to start an already running container MUST have no effect on the container and MUST generate an error.
+Starting a stopped container, one that had been previously be running, MUST create a new process, as specified by the `config.json` file, within the scope of the container.
+
+OPEN: the entire "generate an error" stuff in here is something we should discuss.
+My initial thought was that it would be better to generate an error so that the call would know exactly what happened - meaning, did something new take place or was the container already running. If we didn't return an error (or something) then the caller wouldn't know.  And they can always ignore errors if they want/need to.
+
+### Stop
+
+This operation MUST stop a running container.
+Stopping a container MUST stop all of the processes running within the scope of the container.
```

Comment:
On Tue, Dec 22, 2015 at 11:13:44PM -0800, Brandon Philips wrote:
> > +Stopping a container MUST stop all of the processes running within the scope of the container.
> 
> Weirdness: I put a process into an existing containers namespace but
> not put it into the same cgroup? What is the behavior?

It doesn't get killed.  Unless one of the namespaces it joined was a
PID namespace, and the init for that PID namespace or one of its
ancestors belonged to the cgroup, in which case the kernel's usual
PID-namespace would SIGKILL your process.

That's not weird though, we were clear that stop would only SIGKILL
processes in the given freezer cgroup, and you didn't join the cgroup.
What would you expect to happen?  It's only weird if you don't explain
the stop algorithm, everybody forms conflicting mental images of what
that algorithm is, and then they are surprised when those images don't
match up.